structuralcodes/sections/_generic.py

```diff
@@ -478,6 +581,233 @@ def calculate_moment_curvature(
 
         return res
 
+    def calculate_interaction_domain(
```

Comment:
To be consistent with the other methods, could this be called `calculate_nm_interaction_domain`?